### PR TITLE
Feature/add chunk param to alignment executor

### LINKED ISSUE #1353: ### Range awareness - level2
**TASK: add new parameter to alignment_executor_two_way constructor for chunking**

- [ ] Depends on [only one constructor for alignment_executor_two_way](https://github.com/seqan/seqan3/projects/10#card-27977457)
- [ ] add new chunking parameter `alignment_executor_two_way{sequence_pair_range,algorithm, execution_policy[, chunking = 1]}`
- [ ] test new parameter interface
- [ ] add documentation
- [ ] (NOTE: This does not require actual chunking, only adapt the (public) API)

### PR DESCRIPTION
**Pull request by rrahn**
Depends on #1350 

Fixes #1353

### CODE REVIEW

File: test/unit/alignment/pairwise/execution/alignment_executor_two_way_test.cpp [hunk 1]
```diff
@@ -38,50 +38,84 @@ struct dummy_alignment
...
 struct alignment_executor_two_way_test : public ::testing::Test
-{};
+{
+    // Some globally defined test types
+    using seq_type = all_view<std::string &>;
+    using sequence_pair_t = std::pair<std::string, std::string>;
+    using sequence_pairs_t = std::vector<sequence_pair_t>;
+    using algorithm_t = std::function<size_t(size_t const, seq_type, seq_type)>;
+
+    sequence_pair_t sequence_pair{std::string{"AACGTACGT"}, std::string{"ATCGTCCGT"}};
```
Comment (eseiler) [inline test/unit/alignment/pairwise/execution/alignment_executor_two_way_test.cpp]:
```suggestion
    sequence_pair_t sequence_pair{"AACGTACGT", "ATCGTCCGT"};
```

File: include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp [hunk 2]
```diff
@@ -134,49 +134,46 @@ class alignment_executor_two_way
...
     alignment_executor_two_way(resource_t resrc,
-                               alignment_algorithm_t fn) :
+                               alignment_algorithm_t fn,
+                               size_t chunk_size = 1,
+                               exec_policy_t const & SEQAN3_DOXYGEN_ONLY(exec) = seq) :
         resource{views::zip(std::forward<resource_t>(resrc), std::views::iota(0))},
         resource_it{resource.begin()},
-        kernel{std::move(fn)}
+        kernel{std::move(fn)},
+        _chunk_size(chunk_size)
```
Comment (smehringer) [inline include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp]:
```suggestion
        _chunk_size{chunk_size}
```
:nail_care:

File: include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp [hunk 3]
```diff
@@ -134,49 +134,46 @@ class alignment_executor_two_way
...
     {
+        static_assert(!std::same_as<exec_policy_t, parallel_unsequenced_policy>,
+                      "Parallel unsequenced execution not supported!");
+        static_assert(!std::same_as<exec_policy_t, unsequenced_policy>, "Unsequenced execution not supported!");
+
+        if (chunk_size == 0)
```
Comment (smehringer) [inline include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp]:
```suggestion
        if (chunk_size == 0u)
```

File: include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp [hunk 4]
```diff
@@ -137,8 +137,11 @@ class alignment_executor_two_way
      * \tparam exec_policy_t The type of the execution policy; seqan3::is_execution_policy must return `true`. Defaults
      *                       to seqan3::sequenced_policy.
      * \param[in] resrc The underlying resource containing the sequence pairs to align.
-     * \param[in] fn    The alignment kernel to invoke on the sequences pairs.
-     * \param[in] exec  Optional execution policy to use. Defaults to seqan3::seq.
+     * \param[in] fn The alignment kernel to invoke on the sequences pairs.
+     * \param[in] chunk_size The number of sequences pairs to invoke the alignment algorithm with.
```
Comment (smehringer) [inline include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp]:
```suggestion
     * \param[in] chunk_size The number of sequence pairs to invoke the alignment algorithm with.
```
?

**Pull request merged**